src/status_im2/contexts/onboarding/create_password/view.cljs

```diff
@@ -1,42 +1,197 @@
-
-(defn navigation-bar
+  (:require
+    [quo2.core :as quo]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [status-im2.common.resources :as resources]
+    [status-im2.contexts.onboarding.common.navigation-bar.view :refer [navigation-bar]]
+    [status-im2.contexts.onboarding.create-password.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn background-image
   []
-  [rn/view {:style style/navigation-bar}
-   [quo/page-nav
-    {:align-mid?            true
-     :mid-section           {:type :text-only :main-text ""}
-     :left-section          {:type                :blur-bg
-                             :icon                :i/arrow-left
-                             :icon-override-theme :dark
-                             :on-press            #(rf/dispatch [:navigate-back])}
-     :right-section-buttons [{:type                :blur-bg
-                              :icon                :i/info
-                              :icon-override-theme :dark
-                              :on-press            #(js/alert "Pending")}]}]])
-
-(defn page
+  [rn/image
+   {:style       style/image-background
+    :blur-radius 13
+    :source      (resources/get-image :onboarding-blur-bg)}])
+
+(defn header
   []
-  [rn/view {:style style/page-container}
-   [navigation-bar]
-   [rn/view {:style {:padding-horizontal 20}}
-    [quo/text
-     {:size   :heading-1
-      :weight :semi-bold
-      :style  {:color colors/white}} "Create profile password"]
-    [quo/button
-     {:on-press #(rf/dispatch [:navigate-to :enable-biometrics])
-      :style    {}} (i18n/label :t/continue)]]])
+  [rn/view {:style style/heading}
+   [quo/text
+    {:style  style/heading-title
+     :weight :semi-bold
+     :size   :heading-1}
+    (i18n/label :t/password-creation-title)]
+   [quo/text
+    {:style  style/heading-subtitle
+     :weight :regular
+     :size   :paragraph-1}
+    (i18n/label :t/password-creation-subtitle)]])
+
+(defn password-with-hint
+  [{{:keys [text status shown]} :hint :as input-props}]
+  [rn/view
+   [quo/input
+    (-> input-props
+        (dissoc :hint)
+        (assoc :type           :password
+               :override-theme :dark
+               :blur?          true))]
+   [rn/view {:style style/label-container}
+    (when shown
+      [:<>
+       [quo/icon (if (= status :success) :i/check-circle :i/info)
+        {:container-style style/label-icon
+         :color           (style/label-icon-color status)
+         :size            16}]
+       [quo/text
+        {:style (style/label-color status)
+         :size  :paragraph-2}
+        text]])]])
+
+(defn password-inputs
+  [{:keys [passwords-match? on-change-password on-change-repeat-password on-input-focus
+           password-long-enough? empty-password? same-password-lenght?]}]
+  (reagent/with-let [show-hint-2? (reagent/atom false)]
+    (let [_ (when same-password-lenght? (reset! show-hint-2? true))
```

Comment:
this could be moved outside of let, but still, im wondering why this needed looks suspicious,

Reply:
also typo same-password-lenght? -> same-password-length?